src/components/common/GProfilerModal.tsx

```diff
@@ -0,0 +1,462 @@
+        if (gProfilerLink === null || typeof gProfilerLink !== 'string') return;
+        window.open(gProfilerLink);
+    };
+
+    buildMetricTable = () => {
+        return this.featureMetadata.genes.map((gene: string, idx: number) => {
+            return {
+                gene: gene,
+                ...this.featureMetadata.metrics.reduce(
+                    (accumalatedMetrics, metric) => ({
+                        ...accumalatedMetrics,
+                        [metric.accessor]: metric.values[idx],
+                    }),
+                    {}
+                ),
+            };
+        });
+    };
+
+    buildFeatureQuery = (topNumFeatures, sortedFeatureMetricTable) => {
+        return topNumFeatures
+            .map((topNumFeaturesElement) => {
+                return {
+                    numFeatures: topNumFeaturesElement,
+                    topSortedFeatures: sortedFeatureMetricTable
+                        .slice(0, topNumFeaturesElement)
+                        .reduce(
+                            (acc, sortedFeatureMetricTableRow) => [
+                                ...acc,
+                                sortedFeatureMetricTableRow['gene'],
+                            ],
+                            []
+                        )
+                        .join('\n'),
+                };
+            })
+            .reduce((acc, el) => {
+                return (
+                    acc +
+                    `>Top_${el.numFeatures}` +
+                    '\n' +
+                    el.topSortedFeatures +
+                    '\n'
+                );
+            }, '');
+    };
+
+    buildGProfilerLink = (organism: string, query: string) => {
+        const gProfilerQueryData = {
+            organism: organism,
+            query: query,
+            ordered: 'true',
+            all_results: 'false',
+            no_iea: 'false',
+            combined: 'true',
+            measure_underrepresentation: 'false',
+            domain_scope: 'annotated',
+            significance_threshold_method: 'g_SCS',
+            user_threshold: '0.05',
+            numeric_namespace: 'ENTREZGENE_ACC',
+            sources: 'GO:MF,GO:CC,GO:BP,KEGG,TF,REAC,MIRNA,HPA,CORUM,HP,WP',
+            background: '',
+        };
+
+        const gProfilerQueryString = Object.keys(gProfilerQueryData)
+            .map((key) => {
+                return (
+                    encodeURIComponent(key) +
+                    '=' +
+                    encodeURIComponent(gProfilerQueryData[key])
+                );
+            })
+            .join('&');
+
+        return 'https://biit.cs.ut.ee/gprofiler/gost?' + gProfilerQueryString;
+    };
+
+    handleClickCreateGProfilerLink = async () => {
+        const {
+            topNumFeatures,
+            gProfilerToken,
+            selectedOrganism,
+            selectedSortBy,
+        } = this.state;
+        if (selectedSortBy === null) {
```

Comment:
This should not be possible. Use typescript to ensure `selectedSortBy` is never null (a non-nullable string?) and use a default value in the UI control to ensure there's always a value selected.